internal/venice-client-common/src/main/java/com/linkedin/venice/stats/metrics/MetricEntityState.java

```diff
@@ -0,0 +1,135 @@
+    this.tehutiSensors = tehutiSensors;
+  }
+
+  /**
+   * Add tehuti sensor to tehutiSensors map and throw exception if sensor with same name already exists
+   */
+  public void addTehutiSensors(String name, Sensor tehutiSensor) {
+    if (tehutiSensors == null) {
+      tehutiSensors = new HashMap<>();
+    }
+    if (tehutiSensors.put(name, tehutiSensor) != null) {
+      throw new IllegalArgumentException("Sensor with name '" + name + "' already exists.");
+    }
+  }
+
+  /**
+   * create the metrics/Sensors
+   */
+  @FunctionalInterface
+  public interface TehutiSensorRegistrationFunction {
+    Sensor register(String sensorName, MeasurableStat... stats);
+  }
+
+  public void createMetric(
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      Map<String, List<MeasurableStat>> tehutiMetricInput,
+      TehutiSensorRegistrationFunction registerTehutiSensor) {
+    // Otel metric
+    if (otelRepository != null) {
+      setOtelMetric(otelRepository.createInstrument(this.metricEntity));
+    }
+    // tehuti metric
+    // loop through tehutiMetricInput and call registerTehutiSensor for each String, List<MeasurableStat> pair
+    for (Map.Entry<String, List<MeasurableStat>> entry: tehutiMetricInput.entrySet()) {
+      if (entry.getValue().contains(Percentiles.class)) {
+        addTehutiSensors(
+            entry.getKey(),
+            registerTehutiSensor.register(entry.getKey(), entry.getValue().toArray(new MeasurableStat[0])));
+
+      } else {
+        addTehutiSensors(
+            entry.getKey(),
+            registerTehutiSensor.register(entry.getKey(), entry.getValue().toArray(new MeasurableStat[0])));
+      }
+    }
+  }
+
+  /**
+   * Record otel metrics
+   */
+  private void recordOtelMetric(double value, Attributes otelDimensions) {
+    if (otelMetric != null) {
+      MetricType metricType = this.metricEntity.getMetricType();
+      switch (metricType) {
+        case HISTOGRAM:
+        case MIN_MAX_COUNT_SUM_AGGREGATIONS:
+          ((DoubleHistogram) otelMetric).record(value, otelDimensions);
+          break;
+        case COUNTER:
+          ((LongCounter) otelMetric).add((long) value, otelDimensions);
+          break;
+
+        default:
+          throw new IllegalArgumentException("Unsupported metric type: " + metricType);
+      }
+    }
+  }
+
+  private void recordTehutiMetric(String tehutiMetricName, double value) {
+    if (tehutiSensors != null) {
+      Sensor sensor = tehutiSensors.get(tehutiMetricName);
+      if (sensor != null) {
+        sensor.record(value);
+      }
+    }
+  }
+
+  public void record(String tehutiMetricName, long value, Attributes otelDimensions) {
+    recordOtelMetric(value, otelDimensions);
+    recordTehutiMetric(tehutiMetricName, value);
+  }
+
+  public void record(String tehutiMetricName, double value, Attributes otelDimensions) {
+    recordOtelMetric(value, otelDimensions);
+    recordTehutiMetric(tehutiMetricName, value);
+  }
+}
```

Comment:
These are being used and the concerns about extra looks for every record, which can be done multiple times for a request in hot path can degrade the perf. Modified it to use `RouterTehutiMetricNameEnum` instead of String as a quick change here. Will look into this further for v2.